src/jit/hwintrinsicxarch.cpp

```diff
@@ -482,6 +483,12 @@ GenTree* Compiler::impHWIntrinsic(NamedIntrinsic        intrinsic,
         }
     }
 
+    // TODO-XArch-CQ: Conditionally set `compFloatingPointUsed` based on a flag
+
+    // Set `compFloatingPointUsed` to cover the scenario where an intrinsic is being on SIMD fields, but
+    // where no SIMD local vars are in use. This is the same logic as is used for FEATURE_SIMD.
+    compFloatingPointUsed = true;
```

Comment:
FYI. @CarolEidt, @fiigii 

This is the same logic as https://github.com/dotnet/coreclr/blob/master/src/jit/simd.cpp#L3094. An assert was being hit in LSRA for reflection calls which used `LoadVector128`.

This needs to be cleaned up to use a flag so we aren't setting it on intrinsics which don't actually use any floating-point nodes/registers (such as Crc32).

Also FYI. @sdmaclea. This appears to impact ARM64 as well (based on the FEATURE_SIMD code), and you may need something similar.

Reply:
I also imagine various code could be updated elsewhere in the JIT to properly support TYP_SIMD as floating-point registers, rather than requiring this special handling here.